third_party/validator/cai.go

```diff
@@ -34,3 +40,35 @@ type IAMBinding struct {
 	Role    string   `json:"role"`
 	Members []string `json:"members"`
 }
+
+// assetName templates an asset.name by looking up and replacing all instances
+// of {{field}}. In the case where a field would resolve to an empty string, a
+// generated unique string will be used: "<placeholder-" + randomString() + ">".
+// This is done to preserve uniqueness of asset.name for a given asset.asset_type.
+func assetName(d TerraformResourceData, config *Config, linkTmpl string) (string, error) {
+	re := regexp.MustCompile("{{([[:word:]]+)}}")
+
+	f, err := buildReplacementFunc(re, d, config, linkTmpl)
+	if err != nil {
+		return "", err
+	}
+
+	fWithPlaceholder := func(key string) string {
+		val := f(key)
+		if val == "" {
+			val = fmt.Sprintf("<placeholder-%s>", randString(8))
```

Comment:
Many 'names' in GCP don't allow special characters. I'm guessing you only have to comply with the asset API so it might not be an issue as long as it supports them. But if you ever need to iterate through all things with 'names' and are using those to query GCP apis to check for existence then it could be an issue.

Reply:
I'd suggest we drop the `<`.

Reply:
Removed brackets